### PR TITLE
[Docs] Remove stack trace reference

### PR DESCRIPTION
**Pull request by ErikSchierboom**
I think it might make sense to remove it here. If track's want to add it, they can, but I feel like we should not necessarily encourage including stack traces.

### CODE REVIEW

**Comment by iHiD** [comment]
I think we should see what a load of tracks do and then edit this documentation with some good examples of what works. I really think some consistency is important, but I don't yet understand the consistent thing that we should do.

**Comment by ErikSchierboom** [comment]
I understand, but my thought was not to encourage using stack traces. IMHO it should be a last resort, due the problems with it (filtering unneeded lines, etc.).

File: docs/interface.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ The following overall statuses are valid:
 
 #### Message
 
-Where the status is `error` (the tests fail to execute cleanly), the top level `message` key should be provided. It should provide the occurring error to the user. As it is the only piece of information a user will receive on how to debug their issue, it must be as clear as possible.. For example, in Ruby, in the case of a syntax error, we provide the error and stack trace. In compiled languages, the compilation error should be provided.
+Where the status is `error` (the tests fail to execute cleanly), the top level `message` key should be provided. It should provide the occurring error to the user. As it is the only piece of information a user will receive on how to debug their issue, it must be as clear as possible.. For example, in Ruby, in the case of a syntax error, we provide the error. In compiled languages, the compilation error should be provided.
```
Comment (kotp) [inline docs/interface.md:43]:
```suggestion
Where the status is `error` (the tests fail to execute cleanly), the top level `message` key should be provided. It should provide the occurring error to the user. As it is the only piece of information a user will receive on how to debug their issue, it must be as clear as possible.. For example, in Ruby, in the case of a syntax error, we provide the error without a stack trace. In compiled languages, the compilation error should be provided.
```

**Review by kotp — CHANGES_REQUESTED** [review]
This will keep it explicit that the stack trace is not encouraged, in fact discouraged.

**Comment by iHiD** [comment]
@kotp Thanks for reinitiating the discussion.

---

I don't know where your thinking on this has got to @ErikSchierboom?

I think Stack Traces are really important but only in error cases (either exceptions or compilation errors) and that we remove reference to the code files etc. That's where I'd edge too.

**Comment by ErikSchierboom** [comment]
> I think Stack Traces are really important but only in error cases (either exceptions or compilation errors) and that we remove reference to the code files etc. That's where I'd edge too.

I think stack traces are only important if the default error message is not enough for the student to figure things out. If somehow this is the case, we could present a trimmed, sanitized stack trace, but I'd prefer this to be the last resort. It is all too easy for a stack trace to include sensitive information that the sanitation process does not catch. So I'd argue for a: don't use unless you have a very good reason to do so. The reason being that it makes things easier is not good enough for me, if the student should also be able to figure things out without the stack trace.

**Comment by iHiD** [comment]
@ErikSchierboom Let's push this to the v3-docs instead.

**Pull request closed without merging**